src/Avalonia.Controls/TopLevel.cs

```diff
@@ -328,6 +331,8 @@ bool IInputRoot.ShowAccessKeys
 
         IRenderTarget IRenderRoot.CreateRenderTarget() => CreateRenderTarget();
 
+        public WindowNotificationManager NotificationManager => _notificationManager;
```

Comment:
Let's make initialization of NotificationManager lazy. I.e., create object (and attach) only after getter was called for the first time.
As I see from the tests, current implementation also affects popups, which has a top level too iirc. Having this initialized for all of them even if not needed is a bit unnecessary.

Other than that looks good.